packages/react-components/react-breadcrumb/docs/Spec.md

```diff
@@ -2,54 +2,476 @@
+| ---------------------------------------------------------------------- | ---------- | ----------------- | --------- |
+| Breadcrumb is a component that indicates the path of the current page  | Breadcrumb | Breadcrumb        | ⚠️        |
+| BreadcrumbItem an actionable item within a Breadcrumb                  |            | BreadcrumbItem    | ❌        |
+| BreadcrumbDivider divides BreadcrumbItem components within Breadcrumb  |            | BreadcrumbDivider | ❌        |
+| An BreadcrumbLink represents a anchor to be used inside the Breadcrumb |            | BreadcrumbLink    | ❌        |
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```jsx
+const BreadcrumbV9Example = props => (
+  <Breadcrumb aria-label="breadcrumb">
+    <BreadcrumbItem as="a" href="#">
+      Home
+    </BreadcrumbItem>
+    <BreadcrumbItem as="a" href="#">
+      Gallery
+    </BreadcrumbItem>
+    <BreadcrumbItem as="a" href="#">
+      About
+    </BreadcrumbItem>
+  </Breadcrumb>
+  </>
+);
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Appearance
 
-## API
+Breadcrumb can be:
+
+- transparent (default)
+- subtle
+
+### BreadcrumbItem variants
+
+- Non-clickable element
+- Button
+- Link
+- Dropdown (added by a user)
+
+### Icon
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+BreadcrumbItem can have an icon with text or just icon.
+
+### Size
+
+Breadcrumb can have the following sizes: `small`, `medium` and `large`.
+
+### Tooltip
+
+Tooltips appear on collapsed menu or truncated name of item.
+
+### Dropdown menu
+
+Dropdown contains collapsed items.
+
+## API
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+![visual anatomy of the Breadcrumb component](./assets/breadcrumb-structure.png)
+
+### Components
+
+| Component          | Purpose                                                                 |
+| ------------------ | ----------------------------------------------------------------------- |
+| Breadcrumb         | Wrapper for the Breadcrumb component. Contains `nav` and `ol` elements. |
+| BreadcrumbDivider  | Divider component                                                       |
+| BreadcrumbItem     | Contains `li` element as a wrapper. Can contain button and link         |
+| BreadcrumbDropdown | Can contain subfolders or hidden items.                                 |
+
+### Breadcrumb
+
+#### Anatomy
+
+![visual anatomy of the Breadcrumb component](./assets/breadcrumb-anatomy.png)
+
+#### DOM
+
+```HTML
+<nav aria-label="breadcrumb">
+  <ol>
+  {children}
+  </ol>
+</nav>
+```
+
+#### Breadcrumb structure:
+
+- root `nav` element
+- list `ol` element
+- `divider` - a BreadcrumbDivider component
```

Comment:
How do you envision `divider` being implemented? Would `Breadcrumb` users need to insert it, e.g.

```jsx
<Breadcrumb>
  <BreadcrumbItem/>
  <BreadcrumbDivider/>
  <BreadcrumbItem/>
</Breadcrumb>
```

Reply:
I was going to manage the divider from the main component. In this case, a user needs only to provide items:
```jsx
<Breadcrumb>
  <BreadcrumbItem/>
  <BreadcrumbItem/>
</Breadcrumb>
```
I was thinking about this approach because in case the user's platform is RTL the divider and breadcrumbs will be added from right to left and the chevron will point to the left side. There will be a way to use a custom divider if the user provides their own divider component. Also, I was thinking about an idea to disable the default divider in case if the user wants to be able to do it this way:
```jsx
<Breadcrumb>
  <BreadcrumbItem/>
  <BreadcrumbDivider/>
  <BreadcrumbItem/>
</Breadcrumb>
```

Reply:
If the divider will be managed without JSX, then it will have to be some kind of slot.

``tsx
<BreadcrumbItem divider={{}}>Foo</BreadcrumbItem>
```

This does not feel as intuitive to me as

```tsx
<BreadcrumbItem/>
<BreadcrumbDivider/>
<BreadcrumbItem/>
```

You'll also be enforcing a convention on users whether dividers should be rendered 'before' or 'after' a breadcrumb item, instead of letting users make that choice by letting them use a JSX element

> in case the user's platform is RTL the divider and breadcrumbs will be added from right to left and the chevron will point to the left side

IMO that is fine, in many of our components we internally use something like this

```tsx
const { dir } = useFluent();

let chevron = <ChevronRight />
if (dir === 'rtl') {
  chevron = <ChevronLeft />
}
```

Also if dividers were a slot then users would never be able to treeshake the divider component out of their bundle because it would _**always**_ be included with BreadcrumbItem

Reply:
- About the divider as a slot, that's how I implemented it. Here's a PR: #27199 I'll update the spec. It was more than 2 weeks on review, that's why I started implementing some parts
- I think there's still a way to use it like this:
```jsx
<BreadcrumbItem/>
<BreadcrumbDivider/>
<BreadcrumbItem/>
```
I wanted to have it as part of the item because it's the default behavior and is more readable.  But if the user wants to disable the divider, it can be done by passing `null` to the `variant` prop.

Reply:
No worries, I understand it took me a while to get to this 😅. It's up to you to make the call in the end. The concern I have is mainly that users will _**always**_ have a divider in their bundle whether they use it or not

Reply:
@ling1726 , I thought I sent my comment :) We had a discussion with Oleksandr, I'm gonna use a Divider as a separate component. I'll need to rewrite a spec completely. I'll let you know when I update it.
@layershifter this is a really old implementation, I'm gonna update today it with the latest changes. No `map` will be used in there :) this was just an experimental implementation. I'll try to update it today so tomorrow we can discuss it

Reply:
Resolving this comment as we agreed on using Divider as a separate component.